src/windows.hpp

```diff
@@ -64,6 +66,40 @@ static inline int poll (struct pollfd *pfd, unsigned long nfds, int timeout)
 #define AI_NUMERICSERV 0x0400
 #endif
 
+//  Need unlink() and rmdir() functions that take utf-8 encoded file path.
+static inline std::wstring utf8_to_utf16 (const char *utf8_string)
+{
+    if (!utf8_string || !*utf8_string)
+        return std::wstring ();
+
+    const int utf16_length =
+      ::MultiByteToWideChar (CP_UTF8, MB_ERR_INVALID_CHARS, utf8_string,
+                             -1, // assume the input string is null-terminated
+                             nullptr, 0);
+    if (utf16_length == 0)
+        return std::wstring ();
+
+    std::vector<wchar_t> utf16_string (utf16_length);
```

Comment:
replace by std::wstring, use .resize(utf16_length), and &result[0] as dest address, then return that string. Example:

static std::wstring WideStringFromString (_In_z_ const char *str)
{
    std::wstring retVal;

    if (str && *str) {
        const int cchSource = static_cast<int> (strlen (str));
        const int cchNeeded =
          MultiByteToWideChar (CP_UTF8, 0, str, cchSource, nullptr, 0);

        if (cchNeeded > 0) {
            retVal.resize (cchNeeded);
            MultiByteToWideChar (CP_UTF8, 0, str, cchSource, &retVal[0],
                                 cchNeeded);
        }
    }

    return retVal;
}